lib/puppet/provider/package/pkgin.rb

```diff
@@ -12,8 +12,8 @@
   def self.parse_pkgin_line(package)
 
     # e.g.
-    #   vim-7.2.446 =        Vim editor (vi clone) without GUI
-    match, name, version, status = *package.match(/(\S+)-(\S+)(?: (=|>|<))?\s+.+$/)
+    #   vim-7.2.446;Vim editor (vi clone) without GUI
+    match, name, version, status = *package.match(/(\S+)-(\S+?);(?:(=|>|<))?.+$/)
```

Comment:
Whoops, you're right.  I keep forgetting that a capture group inside a non-capture group still captures.  That makes sense with regards to status.

Perhaps `/(\S+)-(\S+?)[;\s](=|>|<)?.+$/` then?